### PR TITLE
Add transport port fact

### PR DESCRIPTION
**Pull request by kobybr**
Resolves issue #996

### CODE REVIEW

**Comment by tylerjl** [comment]
Jenkins, test this please

**Comment by kobybr** [comment]
Any word on this being merged?

File: lib/facter/es_facts.rb [hunk 1]
```diff
@@ -110,10 +116,29 @@ def self.run
             end
           end
           add_fact(key_prefix, 'plugins', plugin_names.join(','))
+
+          nodes_data['http']['bound_address'].each { |i| http_bound_addresses << i }
+          nodes_data['transport']['bound_address'].each { |i| transport_bound_addresses << i }
+          transport_publish_addresses << nodes_data['transport']['publish_address']
+          transportports << nodes_data['settings']['transport']['tcp']['port']
```
Comment (tylerjl) [inline lib/facter/es_facts.rb]: Looking at an example response for a node's settings from the node stats endpoint, I see the following under `.settings.transport`:

```
{
  "type": {
    "default": "netty4"
  }
}
```

Am I correct in thinking that without a `tcp` key, accessing that `nil` value with `['port']` will throw an error when trying to access a non-existent key on `nil`?

Comment (kobybr) [reply lib/facter/es_facts.rb]: Good catch!   I've pushed new code with what I believe will handle that.

Comment (kobybr) [reply lib/facter/es_facts.rb]: FYI.   This original pull request was intended add additional facts to support those of us who manage firewall/iptables via puppet.

**Comment by tylerjl** [comment]
Thanks @kobybr - the one failed test is unrelated and the nil check looks safer, so this looks good to merge! :+1:

**Comment by kobybr** [comment]
Thanks!

**Pull request merged**